.github/workflows/ci.yml

```diff
@@ -0,0 +1,35 @@
+name: CI
+
+# On every pull request, but only on push to master
+on:
+  push:
+    branches:
+    - master
+    tags:
+    - '*'
+  pull_request:
+
+jobs:
+  tests:
+    runs-on: ubuntu-latest
+    container: osgeo/gdal:ubuntu-full-${{ matrix.gdal-version }}
+    strategy:
+      matrix:
+        python-version: [3.7, 3.8, 3.9]
+        gdal-version: [3.4.0, 3.3.3, 3.2.3, 3.1.3, 3.0.4]
+
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python-version }}
+
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          python -m pip install -e .["test"]
```

Comment:
Right now it's failing with 

```
python: /lib/x86_64-linux-gnu/libm.so.6: version `GLIBC_2.29' not found (required by /__t/Python/3.7.12/x64/lib/libpython3.7m.so.1.0)
python: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.28' not found (required by /__t/Python/3.7.12/x64/lib/libpython3.7m.so.1.0)
```

There might be some libraries to add because we can use python.